### PR TITLE
issue #1423: wire up show field keys

### PR DESCRIPTION
**Pull request by dgnorton**

### CODE REVIEW

File: database.go [hunk 1]
```diff
@@ -312,8 +312,8 @@ func (m *Measurement) idsForExpr(n *influxql.BinaryExpr) (seriesIDs, bool, influ
 // value should be included in the resulting set, and an expression if the return is a field expression.
 // The map that it takes maps each series id to the field expression that should be used to evaluate it when iterating over its cursor.
 // Series that have no field expressions won't be in the map
-func (m *Measurement) walkWhereForSeriesIds(node influxql.Node, filters map[uint32]influxql.Expr) (seriesIDs, bool, influxql.Expr) {
-	switch n := node.(type) {
+func (m *Measurement) walkWhereForSeriesIds(expr influxql.Expr, filters map[uint32]influxql.Expr) (seriesIDs, bool, influxql.Expr) {
```
Comment (otoolep) [inline database.go]: Why the type change?

File: database.go [hunk 2]
```diff
@@ -1176,6 +1194,7 @@ func (db *database) Measurements() Measurements {
 	for _, m := range db.measurements {
 		measurements = append(measurements, m)
 	}
+	sort.Sort(measurements)
```
Comment (otoolep) [inline database.go]: This just to make it look better?

File: influxql/ast.go [hunk 3]
```diff
@@ -69,7 +69,6 @@ func (*GrantStatement) node()                 {}
 func (*ShowContinuousQueriesStatement) node() {}
 func (*ShowDatabasesStatement) node()         {}
 func (*ShowFieldKeysStatement) node()         {}
-func (*ShowFieldValuesStatement) node()       {}
```
Comment (otoolep) [inline influxql/ast.go]: Are we nuking this statement? Just to be sure.

File: httpd/handler_test.go [hunk 4]
```diff
@@ -1520,6 +1520,112 @@ func TestHandler_serveShowTagValues(t *testing.T) {
...
+		{"name": "cpu", "tags": {"host": "server01"},"timestamp": "2009-11-10T23:00:00Z","values": {"field1": 100}},
+		{"name": "cpu", "tags": {"host": "server01", "region": "uswest"},"timestamp": "2009-11-10T23:00:00Z","values": {"field1": 200, "field2": 300, "field3": 400}},
+		{"name": "cpu", "tags": {"host": "server01", "region": "useast"},"timestamp": "2009-11-10T23:00:00Z","values": {"field1": 200, "field2": 300, "field3": 400}},
+		{"name": "cpu", "tags": {"host": "server02", "region": "useast"},"timestamp": "2009-11-10T23:00:00Z","values": {"field1": 200, "field2": 300, "field3": 400}},
+		{"name": "gpu", "tags": {"host": "server01", "region": "useast"},"timestamp": "2009-11-10T23:00:00Z","values": {"field4": 200, "field5": 300}}
+		]}`)
+
+	if status != http.StatusOK {
+		t.Log(body)
+		t.Fatalf("unexpected status after write: %d", status)
```
Comment (otoolep) [inline httpd/handler_test.go]: Can you log the body too? We're seeing some strange failures on Travis, so I'd like to see the reponse body too, in the event this write fails.

File: httpd/handler_test.go [hunk 5]
```diff
@@ -1520,6 +1520,112 @@ func TestHandler_serveShowTagValues(t *testing.T) {
...
+			},
+		},
+	}
+	for i, tt := range tests {
+		query := map[string]string{"db": "foo", "q": tt.q}
+		status, body = MustHTTP("GET", s.URL+`/query`, query, nil, "")
+
+		if status != http.StatusOK {
+			t.Logf("query #%d: %s", i, tt.q)
+			t.Log(body)
```
Comment (otoolep) [inline httpd/handler_test.go]: Great, logging body here.

**Comment by otoolep** [comment]
Generally makes sense, I have some small feedback and a couple of questions.

File: httpd/handler_test.go (hunk 4 above)
Comment (otoolep) [inline httpd/handler_test.go]: Scrap that -- you log it in the line immediately before it. I missed that.

**Comment by benbjohnson** [comment]
:+1:

File: database.go (hunk 1 above)
Comment (dgnorton) [inline database.go]: This function is `walkWhere...` meaning "walk the where clause".  A where clause is of type `influxql.Expr` and it's not valid to pass in any other AST node type.  This change makes that clearer and allows better compile time type checking.

File: database.go (hunk 2 above)
Comment (dgnorton) [inline database.go]: I probably shouldn't have added that.  No need to impose that performance penalty on all callers.

File: influxql/ast.go (hunk 3 above)
Comment (dgnorton) [inline influxql/ast.go]: Yes.  Field values aren't indexed, so we can't do `ShowFieldValues`.

**Pull request merged**